API/Controllers/PortfolioController.cs

```diff
@@ -0,0 +1,498 @@
+        private readonly IProjectService projectService;
+        private readonly IUserService userService;
+        private readonly IAuthorizationHelper authorizationHelper;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PortfolioController"/> class
+        /// </summary>
+        /// <param name="mapper">The mapper which is used to convert the resources to the models to the resource results.</param>
+        /// <param name="portfolioService">The portfolio service which is used to communicate with the logic layer.</param>
+        /// <param name="portfolioItemService">The portfolioItem service which is used to communicate with the logic layer.</param>
+        /// <param name="projectService">The project service which is used to communicate with the logic layer.</param>
+        /// <param name="userService">The user service which is used to communicate with the logic layer.</param>
+        /// <param name="authorizationHelper">The authorization helper which is used to communicate with the authorization helper class.</param>
+        public PortfolioController(IMapper mapper,
+                                   IPortfolioService portfolioService,
+                                   IPortfolioItemService portfolioItemService,
+                                   IProjectService projectService,
+                                   IUserService userService,
+                                   IAuthorizationHelper authorizationHelper)
+        {
+            this.mapper = mapper;
+            this.portfolioService = portfolioService;
+            this.portfolioItemService = portfolioItemService;
+            this.projectService = projectService;
+            this.userService = userService;
+            this.authorizationHelper = authorizationHelper;
+        }
+
+        #region Portfolio
+
+        /// <summary>
+        /// This method is responsible for retrieving a single portfolio.
+        /// </summary>
+        /// <param name="portfolioId">the portfolio identifier which is used for searching a portfolio.</param>
+        /// <returns>This method returns the user resource result.</returns>
+        /// <response code="200">This endpoint returns the portfolio with the specified portfolio id.</response>
+        /// <response code="400">The 400 Bad Request status code is returned when the portfolio id is invalid.</response>
+        /// <response code="404">The 404 Not Found status code is returned when the portfolio with the specified id could not be found.</response>
+        [HttpGet("{portfolioId}")]
+        [ProducesResponseType(typeof(PortfolioResourceResult), (int) HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetPortfolio(int portfolioId)
+        {
+            if(portfolioId < 0)
```

Comment:
I think, an id should not be zero. (If you agree, you should also change the detail property from the problem object.)

```suggestion
            if(portfolioId <= 0)
```